corehq/apps/receiverwrapper/rate_limiter.py

```diff
@@ -88,23 +89,21 @@ def report_submission_usage(domain):
     _report_current_global_submission_thresholds()
 
 
-def _report_rate_limit_submission(domain):
-    datadog_counter('commcare.xform_submissions.rate_limited', tags=[
-        'domain:{}'.format(domain),
-    ])
-
-
-def _delay_and_report_rate_limit_submission_test(domain, max_wait):
+def _delay_and_report_rate_limit_submission(domain, max_wait, datadog_metric):
     with TimingContext() as timer:
-        acquired = submission_rate_limiter.wait(domain, timeout=max_wait)
+        acquired = submission_rate_limiter.wait(domain, timeout=max_wait,
+                                                windows_not_to_wait_on=('hour', 'day', 'week'))
```

Comment:
`('hour', 'day', 'week')` is the default, right?

https://github.com/dimagi/commcare-hq/blob/1c1679314629385ed7dd7fc684856a4e14d6a0e1/corehq/project_limits/rate_limiter.py#L73

I'm curious why you're passing it here?

FWIW, I didn't really understand why you were originally passing `('second', 'minute')` before https://github.com/dimagi/commcare-hq/commit/1c1679314629385ed7dd7fc684856a4e14d6a0e1, but I glossed over it in my review thinking you had some good reason that I was not understanding. Is this parameter ever used to pass anything other than the default? If not, maybe it should be removed? In my view extra parameters like this make the code more complicated and harder to reason about.

Reply:
I went ahead and removed the explicit kwarg in this call, because it was passing in the default value, and I agree that's redundant/confusing.

I had originally in uncommitted code called this kwarg `windows_to_wait_on` with the inverted meaning, and when I changed it I forgot to change `('second', 'minute')` in this call to `('hour', 'day', 'week')` (which is an example of how passing in a kwarg you don't need to is can be error prone).

Semantically, this kwarg (`windows_not_to_wait_on`) means "If the usage has it its limits for any of the following windows, reject it quickly and don't bother waiting". The idea is that if you've already used your hour's allotment, it's very unlikely that waiting 15 seconds is going to get you under the limit; whereas if you're over your second-window allotment, you may very well just be able to wait one second and be under the limit again. (The minute window is somewhere in between, and I decided to wait on it as well.)